src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3899,4 +3816,150 @@ bool ViewProviderSketch::isInEditMode() const
+                    if (isLineSegment(*geo)) {
+                        ++selectedLines;
+                    }
+                    else {
+                        ++selectedConics;
+                    }
+                }
+            }
+            else if (name.substr(0, 4) == "Vert") {
+                ++selectedPoints;
+            }
+            else if (name.substr(0, 4) == "Cons") {
+                ++selectedConstraints;
+            }
+        }
+
+        if (selectedEdges >= 1 && selectedPoints == 0) {
+            menu << "Sketcher_Dimension";
+            if (selectedConics == 0) {
+                menu << "Sketcher_ConstrainHorVer"
+                << "Sketcher_ConstrainVertical"
+                << "Sketcher_ConstrainHorizontal";
+            }
+            else if (selectedConics > 1 && selectedLines == 0){
+                menu << "Sketcher_ConstrainCoincidentUnified"
+                    << "Sketcher_ConstrainTangent"
+                    << "Sketcher_ConstrainEqual";
+            }
+            else if (selectedConics == 1 && selectedLines == 1){
+                menu << "Sketcher_ConstrainTangent";
+            }
+            if (selectedLines > 1 && selectedConics == 0) {
+                menu << "Sketcher_ConstrainParallel"
+                    << "Sketcher_ConstrainPerpendicular"
+                    << "Sketcher_ConstrainTangent"
+                    << "Sketcher_ConstrainEqual";
+            }
+        }
+        else if (selectedEdges == 1 && selectedPoints >= 1) {
+            menu << "Sketcher_Dimension";
+            if (selectedConics == 0) {
+                menu << "Sketcher_ConstrainCoincidentUnified"
+                    << "Sketcher_ConstrainHorVer"
+                    << "Sketcher_ConstrainVertical"
+                    << "Sketcher_ConstrainHorizontal";
+                if (selectedPoints == 2) {
+                    menu << "Sketcher_ConstrainSymmetric";
+                }
+            }
+            else {
+                menu << "Sketcher_ConstrainCoincidentUnified"
+                    << "Sketcher_ConstrainPerpendicular"
+                    << "Sketcher_ConstrainTangent";
+            }
+        }
+        else if (selectedEdges == 0 && selectedPoints >= 1) {
+                menu << "Sketcher_Dimension";
+
+                if (selectedPoints > 1) {
+                menu << "Sketcher_ConstrainCoincidentUnified"
+                    << "Sketcher_ConstrainHorVer"
+                    << "Sketcher_ConstrainVertical"
+                    << "Sketcher_ConstrainHorizontal";
+                }
+        }
+        else if (selectedConstraints >= 1) {
+            menu << "Sketcher_ToggleDrivingConstraint"
+                << "Sketcher_ToggleActiveConstraint"
+                << "Sketcher_SelectElementsAssociatedWithConstraints"
+                << "Separator"
+                << "Std_Delete";
+        }
+        if (selectedConstraints == 0){
```

Comment:
here use this instead so that it covers (1 line 1 constraint) for example : 
`if (selectedPoints != 0 or selectedLines !=0){`

Reply:
changed it, will push when the failing GitHub actions are resolved or should I push every change?

Reply:
I suggest squashing all commits and force push.

Reply:
should I rebase from current main first? How to squash & force push together?

Reply:
I'll give you a hand let's meet : 
https://meet.google.com/fax-ukcg-wkq